pkg/controller/eventlog/eventlog.go

```diff
@@ -0,0 +1,52 @@
+package eventlog
+
+import (
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type ControllerWithEvents interface {
+	EventRecorder() *record.EventRecorder
+	Logger() *logr.Logger
+	Client() *client.Client
+}
+
+type ControllerWithLogger interface {
+	Name() string
```

Comment:
Requiring all controllers implement a `Name()` function seems very verbose just to avoid passing a `name` parameter in to `ValuedLogger` function.

How about just having:

```go
func ValuedLogger(name string, cwl ControllerWithLogger, keysAndValues ...interface{}) logr.Logger {
...
}
```

Reply:
mmm, I don't remember why I did that... need to review

Reply:
done